### DISCUSSION TITLE
Corn and Seconds results are inconsistent

### DISCUSSION THREAD

**Discussion by Chen840302** [go-co-op/gocron#183] [Q&A] [answered]
I have a job1 that is executed every three seconds and takes 5 seconds to execute. 
When the job1 is executed, I will delete job1 from the schedule, 
and when finished the job1, 
I will re-add it to the schedule.

```javascript
func (job *Job)task()  {
	job.Status = "Running"
	log.Println( "RUN", job.Name)
	time.Sleep(time.Second*5)

	job.Status = "Done"
	log.Println(job.Status)
}


func main() {
	log.Println("Starting...")

	tagMap := make(map[string]*Job)
	job1 := Job{Name: "job1", Tag: "job1", Cron: "*/3 * * * * *",Status: "Done"}
	tagMap["job1"] = &job1

	s := gocron.NewScheduler(time.UTC)
	s.WaitForScheduleAll()
	s.StartAsync()

	for{
		time.Sleep(time.Second)
		for k, v := range tagMap{
			switch v.Status {
			case "Running":
				s.RemoveByTag(k)
			case "Waiting":
				//fmt.Println(v.Status, v.Name)
			case "Done":
				s.CronWithSeconds(v.Cron).Tag(v.Tag).Do(job1.task)
				//s.Every(3).Seconds().Tag(v.Tag).Do(job1.task)
				v.Status = "Waiting"
			}
		}
	}
}
```

I use the seconds function is right, 
but the result of Cron is wrong

This is cron 
![image](https://user-images.githubusercontent.com/68727360/120179458-31d88880-c23d-11eb-886c-17444ee52cde.png)

This is seconds function
![image](https://user-images.githubusercontent.com/68727360/120179550-53d20b00-c23d-11eb-9465-318e8be912cf.png)

**Comment by JohnRoesler** [comment, marked as answer]
Looks like we aren’t handling the wait for scheduled logic in cron! Thanks for reporting

**Reply by JohnRoesler** [reply]
https://github.com/go-co-op/gocron/issues/186 resolved
